invenio_rdm_records/services/generators.py

```diff
@@ -155,6 +155,26 @@ def _condition(self, record, file_key=None, **kwargs):
         return is_file_local
 
 
+# class IfRecordWithExternalDOI(ConditionalGenerator):
```

Comment:
Note: commented line through the PR - the approach using permissions. 
Question for the reviewers: which approach seems better?

Reply:
commented in another PR: I think this approach with permissions is cleaner, however it would be nice to hear from @slint as well :)

Reply:
I agree. Then you could add a bit more logic here to handle RDM_ENABLE_VERSIONS_FOR_EXTERNAL_DOI and pass permissions for both can_draft_modify_files and draft_show_version_button.